Build is broken since 7.2.0

<!-- Please don't delete this template otherwise your issue will be closed immediately -->
<!-- Before creating an issue please make sure you are using the latest version of webpack. -->

### Bug report

<!-- Please ask questions on discussions, StackOverflow or the webpack Gitter. -->
<!-- https://github.com/webpack/webpack/discussions -->
<!-- https://stackoverflow.com/questions/ask?tags=webpack -->
<!-- https://gitter.im/webpack/webpack -->
<!-- Issues which contain questions or support requests will be closed. -->

### Actual Behavior

I just upgraded from version 7.1.0 to 7.2.0, the `yarn build` command returns the following error:

`Cannot find module 'ts-node'`

### Expected Behavior

The `yarn build `command should work

### How Do We Reproduce?

<!-- A great way to do this is to provide your configuration via a GitHub repository -->
<!-- The most helpful is a minimal reproduction with instructions on how to reproduce -->
<!-- Repositories with too many files or large `webpack.config.js` files are not suitable -->
<!-- Please only add small code snippets directly into this issue -->
<!-- https://gist.github.com is a good place for longer code snippets -->
<!-- If your issue is caused by a plugin or loader, please create an issue on the loader/plugin repository instead -->

### Please paste the results of `npx webpack info` here, and mention other relevant information

```
  System:
    OS: macOS 12.6.2
    CPU: (8) x64 Apple M1
    Memory: 27.75 MB / 16.00 GB
  Binaries:
    Node: 14.21.1 - /var/folders/z8/9hvjpnrd7yn9qk6dmt7mxk300000gn/T/yarn--1680539298628-0.011389713840826765/node
    Yarn: 1.22.19 - /var/folders/z8/9hvjpnrd7yn9qk6dmt7mxk300000gn/T/yarn--1680539298628-0.011389713840826765/yarn
    npm: 8.19.2 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 111.0.5563.146
    Firefox: 110.0.1
    Safari: 16.4
  Packages:
    babel-loader: ^9.1.2 => 9.1.2 
    clean-webpack-plugin: ^4.0.0 => 4.0.0 
    css-loader: ^6.7.3 => 6.7.3 
    file-loader: ^6.2.0 => 6.2.0 
    html-webpack-plugin: ^5.5.0 => 5.5.0 
    imports-loader: ^4.0.1 => 4.0.1 
    postcss-loader: ^7.2.0 => 7.2.0 
    replace-in-file-webpack-plugin: ^1.0.6 => 1.0.6 
    sass-loader: ^13.2.2 => 13.2.2 
    style-loader: ^3.3.2 => 3.3.2 
    terser-webpack-plugin: ^5.3.7 => 5.3.7 
    webpack: ^5.77.0 => 5.77.0 
    webpack-cli: ^5.0.1 => 5.0.1 
    webpack-dev-server: ^4.13.2 => 4.13.2 
    webpack-merge: ^5.8.0 => 5.8.0 
    webpack-notifier: ^1.15.0 => 1.15.0 
```

--- a/src/utils.js
+++ b/src/utils.js
@@ -49,70 +49,76 @@ async function loadConfig(loaderContext, config, postcssOptions) {
     throw new Error(`No PostCSS config found in: ${searchPath}`);
   }
 
+  let isTsNodeInstalled = false;
+
+  try {
+    // eslint-disable-next-line import/no-extraneous-dependencies, global-require
+    require("ts-node");
+
+    isTsNodeInstalled = true;
+  } catch (_) {
+    // Nothing
+  }
+
   const moduleName = "postcss";
-  const explorer = cosmiconfig(moduleName, {
-    searchPlaces: [
-      "package.json",
-      `.${moduleName}rc`,
-      `.${moduleName}rc.json`,
-      `.${moduleName}rc.yaml`,
-      `.${moduleName}rc.yml`,
-      `.${moduleName}rc.js`,
-      `.${moduleName}rc.mjs`,
-      `.${moduleName}rc.cjs`,
-      `.${moduleName}rc.ts`,
-      `.${moduleName}rc.mts`,
-      `.${moduleName}rc.cts`,
-      `.config/${moduleName}rc`,
-      `.config/${moduleName}rc.json`,
-      `.config/${moduleName}rc.yaml`,
-      `.config/${moduleName}rc.yml`,
-      `.config/${moduleName}rc.js`,
-      `.config/${moduleName}rc.mjs`,
-      `.config/${moduleName}rc.cjs`,
-      `.config/${moduleName}rc.ts`,
-      `.config/${moduleName}rc.mts`,
-      `.config/${moduleName}rc.cts`,
-      `${moduleName}.config.js`,
-      `${moduleName}.config.mjs`,
-      `${moduleName}.config.cjs`,
-      `${moduleName}.config.ts`,
-      `${moduleName}.config.mts`,
-      `${moduleName}.config.cts`,
-    ],
-    loaders: {
-      ".js": async (...args) => {
-        let result;
+  const searchPlaces = isTsNodeInstalled
+    ? [
+        "package.json",
+        `.${moduleName}rc`,
+        `.${moduleName}rc.json`,
+        `.${moduleName}rc.yaml`,
+        `.${moduleName}rc.yml`,
+        `.${moduleName}rc.js`,
+        `.${moduleName}rc.mjs`,
+        `.${moduleName}rc.cjs`,
+        `.${moduleName}rc.ts`,
+        `.${moduleName}rc.mts`,
+        `.${moduleName}rc.cts`,
+        `.config/${moduleName}rc`,
+        `.config/${moduleName}rc.json`,
+        `.config/${moduleName}rc.yaml`,
+        `.config/${moduleName}rc.yml`,
+        `.config/${moduleName}rc.js`,
+        `.config/${moduleName}rc.mjs`,
+        `.config/${moduleName}rc.cjs`,
+        `.config/${moduleName}rc.ts`,
+        `.config/${moduleName}rc.mts`,
+        `.config/${moduleName}rc.cts`,
+        `${moduleName}.config.js`,
+        `${moduleName}.config.mjs`,
+        `${moduleName}.config.cjs`,
+        `${moduleName}.config.ts`,
+        `${moduleName}.config.mts`,
+        `${moduleName}.config.cts`,
+      ]
+    : [
+        "package.json",
+        `.${moduleName}rc`,
+        `.${moduleName}rc.json`,
+        `.${moduleName}rc.yaml`,
+        `.${moduleName}rc.yml`,
+        `.${moduleName}rc.js`,
+        `.${moduleName}rc.mjs`,
+        `.${moduleName}rc.cjs`,
+        `.config/${moduleName}rc`,
+        `.config/${moduleName}rc.json`,
+        `.config/${moduleName}rc.yaml`,
+        `.config/${moduleName}rc.yml`,
+        `.config/${moduleName}rc.js`,
+        `.config/${moduleName}rc.mjs`,
+        `.config/${moduleName}rc.cjs`,
+        `${moduleName}.config.js`,
+        `${moduleName}.config.mjs`,
+        `${moduleName}.config.cjs`,
+      ];
 
-        try {
-          result = defaultLoaders[".js"](...args);
-        } catch (error) {
-          let importESM;
+  const loaders = {
+    ".js": async (...args) => {
+      let result;
 
-          try {
-            // eslint-disable-next-line no-new-func
-            importESM = new Function("id", "return import(id);");
-          } catch (e) {
-            importESM = null;
-          }
-
-          if (
-            error.code === "ERR_REQUIRE_ESM" &&
-            url.pathToFileURL &&
-            importESM
-          ) {
-            const urlForConfig = url.pathToFileURL(args[0]);
-
-            result = await importESM(urlForConfig);
-          } else {
-            throw error;
-          }
-        }
-
-        return result;
-      },
-      ".mjs": async (...args) => {
-        let result;
+      try {
+        result = defaultLoaders[".js"](...args);
+      } catch (error) {
         let importESM;
 
         try {
@@ -122,20 +128,53 @@ async function loadConfig(loaderContext, config, postcssOptions) {
           importESM = null;
         }
 
-        if (url.pathToFileURL && importESM) {
+        if (
+          error.code === "ERR_REQUIRE_ESM" &&
+          url.pathToFileURL &&
+          importESM
+        ) {
           const urlForConfig = url.pathToFileURL(args[0]);
 
           result = await importESM(urlForConfig);
         } else {
-          throw new Error("ESM is not supported");
+          throw error;
         }
+      }
 
-        return result;
-      },
-      ".cts": TypeScriptLoader(),
-      ".mts": TypeScriptLoader(),
-      ".ts": TypeScriptLoader(),
+      return result;
     },
+    ".mjs": async (...args) => {
+      let result;
+      let importESM;
+
+      try {
+        // eslint-disable-next-line no-new-func
+        importESM = new Function("id", "return import(id);");
+      } catch (e) {
+        importESM = null;
+      }
+
+      if (url.pathToFileURL && importESM) {
+        const urlForConfig = url.pathToFileURL(args[0]);
+
+        result = await importESM(urlForConfig);
+      } else {
+        throw new Error("ESM is not supported");
+      }
+
+      return result;
+    },
+  };
+
+  if (isTsNodeInstalled) {
+    loaders[".cts"] = TypeScriptLoader();
+    loaders[".mts"] = TypeScriptLoader();
+    loaders[".ts"] = TypeScriptLoader();
+  }
+
+  const explorer = cosmiconfig(moduleName, {
+    searchPlaces,
+    loaders,
   });
 
   let result;

